### ISSUE TITLE
How to get gradients of FrozenBatchNorm2d after loss.backward()?

### ISSUE THREAD

**Issue by zeyu-hello** [facebookresearch/detectron2#1524]
## ❓ How to do something using detectron2

Describe what you want to do, including:
Since I want to get a statistics of gradients of FrozenBatchNorm2d layer in faster-rcnn, I inherit `DefaultTrainer` and reproduce as below:

```
class MyTrainer(DefaultTrainer):

    def run_step(self):
        """
        Implement the standard training logic described above.
        """
        assert self.model.training, "[SimpleTrainer] model was changed to eval mode!"
        start = time.perf_counter()
        """
        If you want to do something with the data, you can wrap the dataloader.
        """
        data = next(self._data_loader_iter)
        data_time = time.perf_counter() - start

        """
        If you want to do something with the losses, you can wrap the model.
        """
        loss_dict = self.model(data)
        losses = sum(loss_dict.values())
        self._detect_anomaly(losses, loss_dict)

        metrics_dict = loss_dict
        metrics_dict["data_time"] = data_time
        self._write_metrics(metrics_dict)

        """
        If you need to accumulate gradients or something similar, you can
        wrap the optimizer with your custom `zero_grad()` method.
        """
        self.optimizer.zero_grad()
        losses.backward()

        """
        If you need to accumulate gradients or something similar, you can
        wrap the optimizer with your custom `zero_grad()` method.
        """
        for layer in self.model.modules():
            if isinstance(layer, FrozenBatchNorm2d):
                print("AAAA: ", layer.weight.grad)      # <<----------------------------- HERE!!!!!!!!
        """
        If you need gradient clipping/scaling or other processing, you can
        wrap the optimizer with your custom `step()` method.
        """
        self.optimizer.step()
```

I got `AAAA: None`. So how do i get gradients.

I noticed it's ok if I change the layer to `nn.Conv2d`.

Thanks!

**Comment by ppwwyyxx** [comment]
The layer does not have gradients by [definition](https://detectron2.readthedocs.io/modules/layers.html#detectron2.layers.FrozenBatchNorm2d), because it's frozen.
